File: ios/VllowDugout/Models/Product.swift
import Foundation

nonisolated struct Product: Identifiable, Hashable, Sendable {
    let id: UUID
    let name: String
    let subtitle: String
    let category: ProductCategory
    let priceCoins: Int
    let imageURL: String
    let woodType: String
    let weight: String
    let balance: String
    let handleGrip: String
    let powerRating: Int
    let description: String

    init(
        id: UUID = UUID(),
        name: String,
        subtitle: String,
        category: ProductCategory,
        priceCoins: Int,
        imageURL: String = "",
        woodType: String = "",
        weight: String = "",
        balance: String = "",
        handleGrip: String = "",
        powerRating: Int = 0,
        description: String = ""
    ) {
        self.id = id
        self.name = name
        self.subtitle = subtitle
        self.category = category
        self.priceCoins = priceCoins
        self.imageURL = imageURL
        self.woodType = woodType
        self.weight = weight
        self.balance = balance
        self.handleGrip = handleGrip
        self.powerRating = powerRating
        self.description = description
    }
}

nonisolated enum ProductCategory: String, CaseIterable, Sendable {
    case all = "All Gear"
    case bats = "Bats"
    case gloves = "Gloves"
    case protective = "Protective"
    case apparel = "Apparel"
}

File: flutter_app/lib/models/product.dart
import 'package:uuid/uuid.dart';

enum ProductCategory {
  all('All Gear'),
  bats('Bats'),
  gloves('Gloves'),
  protective('Protective'),
  apparel('Apparel');

  final String displayName;
  const ProductCategory(this.displayName);
}

class Product {
  final String id;
  final String name;
  final String subtitle;
  final ProductCategory category;
  final int priceCoins;
  final String imageURL;
  final String woodType;
  final String weight;
  final String balance;
  final String handleGrip;
  final int powerRating;
  final String description;

  Product({
    String? id,
    required this.name,
    required this.subtitle,
    required this.category,
    required this.priceCoins,
    this.imageURL = '',
    this.woodType = '',
    this.weight = '',
    this.balance = '',
    this.handleGrip = '',
    this.powerRating = 0,
    this.description = '',
  }) : id = id ?? const Uuid().v4();

  @override
  bool operator ==(Object other) =>
      identical(this, other) || other is Product && id == other.id;

  @override
  int get hashCode => id.hashCode;
}
